### ISSUE TITLE
Example Files for Movie and Real Time modes

### ISSUE THREAD

**Issue by chrisndeca** [scrool/xled#50]
It would be super helpful if you could publish an example movie file that can be uploaded onto the Twinkly lights and played.
 It would also be great if there was a file or directions on how to send real time data to the Twinkly's. 

I am trying to get these lights to run the way I want them to, not based on the lame animations that come with the iPhone App!

Thanks!

**Comment by scrool** [comment]
I thought of that in the past. I could do so only for my single model that I own. And that wouldn't work for any others nor I wanted to publish something that I cannot test.

I'm wondering if you have seen a tried to follow https://xled-docs.readthedocs.io/en/latest/protocol_details.html#upload-full-movie-led-effect ?

**Comment by chrisndeca** [comment]
Well you could put a disclaimer on it that it is just a sample file, for your set, and your milage may vary on other sets. At least it would give someone like me some sort of starting point. It would be great to see what is at the head of a new movie file, what delimiter there is for new frames, that sort of thing. 

I have 2 of the curtain and 2 of the 56 light strings, both version 1's. 

And is the real time data also based on individual sets?

**Comment by scrool** [comment]
> It would be great to see what is at the head of a new movie file, what delimiter there is for new frames, that sort of thing.

There there is no header nor any deliminer.

> And is the real time data also based on individual sets?

Yes. Except that real time data has header that is used to authenticate on its own - since it is separate UDP packet compared to any other communication that uses HTTP and has advantage of HTTP headers where authentication is.

**Comment by chrisndeca** [comment]
So the movie file is just a text file of numbers converted over to binary?
So 175115062
or
001100010011011100110101001100010011000100110101001100000011011000110010
one of those, if sent to the lights, should turn the first light whatever color
R175 G115 B062 is? And for a 56 light string, I would need 55 more sequences like those?

I have tried sending both the text file and the binary file as both a file reference in the code (/home/directory/file) and by opening them into Python and sending that along to the strings, as well as reading both the text and binary into strings, and sending them that way.

So after authenticating and getting a token and verifying that back to the string, I set up some headers
`headers4_plus = {'X-Auth-Token': 'NrAl1Kf5anc=','Content-Type': 'application/octet-stream'}`

and then set up the command
`uploadMovie = requests.post('http://10.0.1.70/xled/v1/led/movie/full','/home/pi/binary-output.bin', headers=headers4_plus); print(uploadMovie.text)`

and what I get back from that is 
`<ApplicationResponse [None]>`
and I believe I am supposed to get something like 
`<ApplicationResponse Playing> `
if successful.
What the heck am I doing wrong?
I obviously have no clue as to what I am doing, so any guidance on this would be much appreciated!

**Comment by scrool** [comment]
> So the movie file is just a text file of numbers converted over to binary?

No. It is binary file. If you would have hypothetical LED strip with single RGB pixel and wanted to set them to values:
red: 175, green: 115 and blue: 62

You'd have to create a file with binary representations of those and end up with a file with just three bytes. Example in python:

```
$ python -c 'from struct import pack; open("/tmp/movie", "wb").write(pack("BBB", 175, 115, 62))'
$ xxd /tmp/movie
00000000: af73 3e                                  .s>
```

> And for a 56 light string, I would need 55 more sequences like those?

Not entirely what you wrote but yes, for LED strip with 56 RGB pixels you have to create a file that is exactly 56*3=168 bytes long.

**Comment by chrisndeca** [comment]
Thank you for your guidance last night. I think I am very very close here. I have the 168 byte file (thanks to your python commands. I am uploading the file to the IP address /xled/v1/led/movie/full
The command waits for about 15-20 seconds, like the file might be uploading, but then I get a bunch of errors. The final error says "File "/usr/lib/python3/dist-packages/requests/adapters.py", line 498, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: Remote end closed connection without response"

Does this mean that python is closing the connection to the lights prematurely?

I have 'Content-Type': 'application/octet-stream' in the headers, so the string should know what is coming.
Here is the file I am using. 
[movie.zip](https://github.com/scrool/xled/files/5595105/movie.zip)

Anything that you can think of that I might be doing wrong?

**Comment by scrool** [comment]
Have you followed this https://xled-docs.readthedocs.io/en/latest/protocol_details.html#upload-full-movie-led-effect ? I believe you are missing last step https://xled-docs.readthedocs.io/en/latest/rest_api.html#set-led-movie-config .

I haven't uploaded those movies for couple of months but I believe uploading movie is not enough.

**Comment by chrisndeca** [comment]
I did it!!!! Thank you so much! 

Attached is the python script that I wrote (It is a mess, sorry, I wrote it to act as a starting point for many functions, not just uploading a movie, and I am pretty much a novice when it comes to Python) that will upload the also enclosed binary frame to a 56 light strand of bulbs (it turns every other light white and every other light off). Feel free to post it with your project here. Hopefully it will save someone else from spending 3 days how to figure this all out (like I said, a novice at python).

I do have one more question that you might be able to help with. It has to do with the small script yo shared earlier that was vital to me figuring out how to format a frame. If you don't want to answer it, I can go elsewhere, just let me know.

And again, thanks!
[TwinklyMovieUpload.zip](https://github.com/scrool/xled/files/5600663/TwinklyMovieUpload.zip)
